internal/provider/provider.go

```diff
@@ -439,7 +439,10 @@ func createOnCallClient(d *schema.ResourceData) (*onCallAPI.Client, error) {
 	return onCallAPI.New(baseURL, aToken)
 }
 
+// Sets a custom HTTP Header on all requests coming from the Grafana Terraform Provider to Grafana-Terraform-Provider: true
+// in addition to any headers set within the `http_headers` field or the `GRAFANA_HTTP_HEADERS` environment variable
 func getHTTPHeadersMap(d *schema.ResourceData) (map[string]string, error) {
+	headers := map[string]string{"Grafana-Terraform-Provider": "true"}
```

Comment:
Sorry to ask after this is already merged, but:
What do you think about making this header more flexible? Maybe something like `"Grafana-Request-Origion": "terraform"`? That lets us provide options for other tooling in the future - we could propose adding headers to grabana and also to tools Grafana provides.

I've raised an issue on grafana/grafana

Reply:
Open to renaming this key-value pair to something more suitable - let's see if anybody has more thoughts on the [discussion](https://github.com/grafana/grafana/discussions/68049) within grafana/grafana that you've linked out below, and I'm open to modifying this.